### ISSUE TITLE
EFCore 3.x or EFClassic 6 for full framework project?

### ISSUE THREAD

**Issue by julealgon** [dotnet/efcore#28941]
## Ask a question

Given the restriction of a .NET 4.8, from a support perspective, would it be best to use EFCore 3.x, or should I just use EF6?

I'm aware that EFCore 3 is not as fully featured as EF6, but I wanted to know which one of these 2 will be more likely to have future support etc. I might need to host an EF dbcontext in a webforms application at least temporarily, so I'd like to know which is best at this moment for that purpose, assuming both versions contain the feature set I need.

**Comment by ErikEJ** [comment]
The support policies are here: https://docs.microsoft.com/en-us/ef/efcore-and-ef6/support - notice that EF Core 3.1 support ends December 2022

**Comment by roji** [comment]
Note that although non-core EF6 is technically still supported, we will likely only make any changes/fixes if a critical security issue is identified. In practice, both 3.x and non-Core 6 are legacy and will not receive either new features or (non-security) bug fixes.

**Comment by ajcvickers** [comment]
@julealgon To add to what Erik and Shay have said, using .NET Framework is a dead-end. It is what it is and is still supported in that you can contact Microsoft Support about it (which may incur a fee). But it won't be getting new features and usually won't be getting bug fixes either. It will get security patches. The same is true for EF6.

EF Core 3.1, on the other hand, is going fully out-of-support at the end of this year. That means no kind of support, and no fixes, even for security.

So, bottom line is, don't use EF Core 3.1. And only use .NET Framework if you are happy with what it is and will never want anything more.
